syntax = "proto3";

package zmk.template;

// Battery history entry
message BatteryHistoryEntry {
    uint32 timestamp = 1;  // Unix timestamp (seconds since epoch)
    uint32 battery_percentage = 2;  // Battery percentage 0-100
}

// Request to get battery history
message GetBatteryHistoryRequest {
    // Empty - returns all stored history
}

// Response with battery history
message GetBatteryHistoryResponse {
    repeated BatteryHistoryEntry entries = 1;
    uint32 current_battery = 2;  // Current battery percentage
    uint32 total_entries = 3;  // Total number of entries stored
}

// Request to clear battery history
message ClearBatteryHistoryRequest {
    // Empty - clears all history
}

// Response after clearing history
message ClearBatteryHistoryResponse {
    bool success = 1;
}

message Request {
    oneof request_type {
        GetBatteryHistoryRequest get_battery_history = 1;
        ClearBatteryHistoryRequest clear_battery_history = 2;
    }
}

message ErrorResponse {
    string message = 1;
}

message Response {
    oneof response_type {
        ErrorResponse error = 1;
        GetBatteryHistoryResponse battery_history = 2;
        ClearBatteryHistoryResponse clear_battery_history = 3;
    }
}
